src/vcpkg/commands.ci.cpp

```diff
@@ -218,51 +255,118 @@ namespace
                                   : SortedVector<std::string>(Strings::split(it_exclusions->second, ',')));
     }
 
-    bool print_regressions(const std::vector<SpecSummary>& results,
-                           const std::map<PackageSpec, BuildResult>& known,
-                           const CiBaselineData& cidata,
+    bool print_regressions(const std::map<PackageSpec, CiResult>& ci_results,
+                           const CiBaselineData& baseline_data,
                            const std::string* ci_baseline_file_name,
-                           const LocalizedString& not_supported_regressions,
                            bool allow_unexpected_passing)
     {
-        bool has_error = !not_supported_regressions.empty();
+        bool has_error = false;
         LocalizedString output = msg::format(msgCiBaselineRegressionHeader);
         output.append_raw('\n');
-        output.append(not_supported_regressions);
-        for (auto&& r : results)
+        for (auto&& ci_result : ci_results)
         {
-            auto result = r.build_result.value_or_exit(VCPKG_LINE_INFO).code;
-            auto msg = format_ci_result(r.get_spec(),
-                                        result,
-                                        cidata,
-                                        ci_baseline_file_name,
-                                        allow_unexpected_passing,
-                                        !r.is_user_requested_install());
+            auto msg = format_ci_result(
+                ci_result.first, ci_result.second.code, baseline_data, ci_baseline_file_name, allow_unexpected_passing);
             if (!msg.empty())
             {
                 has_error = true;
                 output.append(msg).append_raw('\n');
             }
         }
-        for (auto&& r : known)
+
+        if (has_error)
+        {
+            msg::write_unlocalized_text_to_stderr(Color::none, output);
+        }
+
+        return has_error;
+    }
+
+    std::vector<PackageSpec> calculate_packages_with_qualified_deps(
+        const std::vector<const SourceControlFileAndLocation*>& all_control_files, const Triplet& target_triplet)
+    {
+        std::vector<PackageSpec> ret;
+        for (auto scfl : all_control_files)
         {
-            auto msg =
-                format_ci_result(r.first, r.second, cidata, ci_baseline_file_name, allow_unexpected_passing, true);
-            if (!msg.empty())
+            if (scfl->source_control_file->has_qualified_dependencies() ||
+                !scfl->source_control_file->core_paragraph->supports_expression.is_empty())
```

Comment:
This should probably check features as well for qualified dependencies and supports expressions.

But I think this is just a performance optimization, not correctness.

Reply:
I'm not sure but I didn't actually change this so I'm going to avoid changing it. (I just extracted this block as a function:

https://github.com/microsoft/vcpkg-tool/blob/4819c236bb6a9e51be6e46a226683a64b11fc409/src/vcpkg/commands.ci.cpp#L90-L99

)